File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizFilterSheet.kt
package kh.mcnc.lib.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@OptIn(ExperimentalLayoutApi::class)
@Composable
fun BizFilterSheet(
    isVisible: Boolean,
    selectedFacilities: List<String>,
    onDismissRequest: () -> Unit,
    onApply: (List<String>) -> Unit
) {
    val facilities = listOf("Korea Desk", "Japan Desk", "China Desk", "SME Center", "Coffee")
    var currentSelection by remember { mutableStateOf(selectedFacilities) }

    LaunchedEffect(isVisible) {
        if (isVisible) {
            currentSelection = selectedFacilities.toList()
        }
    }

    BizBottomSheet(
        isVisible = isVisible,
        onDismissRequest = onDismissRequest
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 24.dp)
                .padding(bottom = 32.dp)
        ) {
            Text(
                text = "Filter",
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF1A2A5E),
                modifier = Modifier.padding(bottom = 8.dp)
            )
            Text(
                text = "Search branch you wish to visit faster.",
                fontSize = 14.sp,
                color = Color(0xFF555555),
                modifier = Modifier.padding(bottom = 24.dp)
            )

            Text(
                text = "BRANCH FACILITY",
                fontSize = 12.sp,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF1A2A5E),
                letterSpacing = 0.5.sp,
                modifier = Modifier.padding(bottom = 16.dp)
            )

            FlowRow(
                modifier = Modifier.padding(bottom = 32.dp),
                horizontalArrangement = Arrangement.spacedBy(12.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                facilities.forEach { facility ->
                    val isActive = currentSelection.contains(facility)
                    Surface(
                        shape = RoundedCornerShape(20.dp),
                        color = if (isActive) Color(0xFF1A2A5E) else Color.White,
                        border = BorderStroke(1.dp, if (isActive) Color(0xFF1A2A5E) else Color(0xFFEAEAEA)),
                        modifier = Modifier.clickable {
                            currentSelection = if (isActive) {
                                currentSelection - facility
                            } else {
                                currentSelection + facility
                            }
                        }
                    ) {
                        Text(
                            text = facility,
                            color = if (isActive) Color.White else Color(0xFF111111),
                            fontSize = 13.sp,
                            fontWeight = FontWeight.SemiBold,
                            modifier = Modifier.padding(horizontal = 16.dp, vertical = 10.dp)
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.weight(1f, fill = false))

            Column(
                modifier = Modifier.fillMaxWidth(),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                BizButton(
                    text = "Apply",
                    variant = BizButtonVariant.Primary,
                    block = true,
                    onClick = {
                        onApply(currentSelection)
                        onDismissRequest()
                    }
                )
                Surface(
                    shape = RoundedCornerShape(12.dp),
                    color = Color.White,
                    border = BorderStroke(1.dp, Color(0xFFEAEAEA)),
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable { onDismissRequest() }
                ) {
                    Box(
                        modifier = Modifier.padding(14.dp),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = "Close",
                            color = Color(0xFF1A2A5E),
                            fontSize = 15.sp,
                            fontWeight = FontWeight.SemiBold
                        )
                    }
                }
            }
        }
    }
}

File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizFilterSheet.swift
import SwiftUI

public struct BizFilterSheetModifier: ViewModifier {
    @Binding var isPresented: Bool
    @Binding var selectedFacilities: [String]
    var onApply: (([String]) -> Void)?
    
    let facilities = ["Korea Desk", "Japan Desk", "China Desk", "SME Center", "Coffee"]
    
    public func body(content: Content) -> some View {
        content.bizBottomSheet(
            isPresented: $isPresented,
            header: { EmptyView() },
            footer: {
                VStack(spacing: 12) {
                    BizButton("Apply", variant: .primary, action: {
                        onApply?(selectedFacilities)
                        isPresented = false
                    })
                    
                    BizButton("Close", variant: .outline, action: {
                        isPresented = false
                    })
                }
            },
            content: {
                VStack(alignment: .leading, spacing: 0) {
                    Text("Filter")
                        .font(.system(size: 20, weight: .bold))
                        .foregroundColor(Color(hex: "#1A2A5E"))
                        .padding(.bottom, 8)
                    
                    Text("Search branch you wish to visit faster.")
                        .font(.system(size: 14))
                        .foregroundColor(Color(hex: "#555555"))
                        .padding(.bottom, 32)
                    
                    Text("BRANCH FACILITY")
                        .font(.system(size: 12, weight: .bold))
                        .foregroundColor(Color(hex: "#1A2A5E"))
                        .tracking(0.5)
                        .padding(.bottom, 16)
                    
                    // Simple wrapping layout for chips (FlowLayout replacement)
                    // In a production iOS app, this would use a custom layout or UICollectionView
                    // For this framework sample, we will use a LazyVGrid or wrapped HStacks
                    ScrollView {
                        // Wrapping HStack simulation
                        let chunked = facilities.chunked(into: 3)
                        VStack(alignment: .leading, spacing: 12) {
                            ForEach(0..<chunked.count, id: \.self) { rowIndex in
                                HStack(spacing: 12) {
                                    ForEach(chunked[rowIndex], id: \.self) { facility in
                                        FilterChip(
                                            title: facility,
                                            isSelected: selectedFacilities.contains(facility),
                                            action: {
                                                if selectedFacilities.contains(facility) {
                                                    selectedFacilities.removeAll { $0 == facility }
                                                } else {
                                                    selectedFacilities.append(facility)
                                                }
                                            }
                                        )
                                    }
                                }
                            }
                        }
                    }
                    .frame(maxHeight: 150)
                }
                .padding(.horizontal, 24)
                .padding(.top, 8)
            }
        )
    }
}

private struct FilterChip: View {
    let title: String
    let isSelected: Bool
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            Text(title)
                .font(.system(size: 13, weight: .semibold))
                .padding(.horizontal, 16)
                .padding(.vertical, 10)
                .foregroundColor(isSelected ? .white : Color(hex: "#111111"))
                .background(isSelected ? Color(hex: "#1A2A5E") : Color.white)
                .cornerRadius(20)
                .overlay(
                    RoundedRectangle(cornerRadius: 20)
                        .stroke(isSelected ? Color(hex: "#1A2A5E") : Color(hex: "#EAEAEA"), lineWidth: 1)
                )
        }
    }
}

extension Array {
    func chunked(into size: Int) -> [[Element]] {
        return stride(from: 0, to: count, by: size).map {
            Array(self[$0 ..< Swift.min($0 + size, count)])
        }
    }
}

public extension View {
    func bizFilterSheet(
        isPresented: Binding<Bool>,
        selectedFacilities: Binding<[String]>,
        onApply: (([String]) -> Void)? = nil
    ) -> some View {
        self.modifier(BizFilterSheetModifier(
            isPresented: isPresented,
            selectedFacilities: selectedFacilities,
            onApply: onApply
        ))
    }
}
